### DISCUSSION TITLE
I need help with `vim.opt.path` please

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2065] [Q&A] [answered]
In my init.lua config I had this:
```lua
vim.opt.path = vim.opt.path + "**"
```
Can you please help me: how do I do this with nixvim?
I tried various things, like this for example:
```nix
opts.path = path + "**";
```

Thanks In advance.

**Comment by MattSturgeon** [comment, marked as answer]
I think this should be possible by passing "raw lua".

Can you try:

```nix
opt.path.__raw = ''vim.opt.path + "**"'';
```

?

**Reply by refaelsh** [reply]
It worked, yes. Thank you :-)
Is there a way to do it in a "nixvim" way? E.g., without using `__raw`?

**Reply by MattSturgeon** [reply]
Well `__raw` actually _is_ a nixvim feature, and wanting to avoid it entirely is probably an anti-pattern.

The thing is, nix modules are great, but they can't represent every conceivable thing one may wish to write in lua. This is why we provide ways to break through our abstractions and just write actual lua code.

If you're using a recent version of our main branch, you can also use nixpkg's equivalent function `lib.generators.mkLuaInline` which does essentially the same thing as our `mkRaw`, but is actually a part of nixpkgs.

**Reply by refaelsh** [reply]
I understand. Thank you very much!
